src/bytes/complete.rs

```diff
@@ -242,55 +242,26 @@ where
 {
   move |i: I| {
     let input = i;
+    let input_len = input.input_len();
 
-    match input.position(|c| !cond(c)) {
-      Some(idx) => {
-        if idx >= m {
-          if idx <= n {
-            let res: IResult<_, _, Error> = if let Ok(index) = input.slice_index(idx) {
-              Ok(input.take_split(index))
-            } else {
-              Err(Err::Error(Error::from_error_kind(
-                input,
-                ErrorKind::TakeWhileMN,
-              )))
-            };
-            res
-          } else {
-            let res: IResult<_, _, Error> = if let Ok(index) = input.slice_index(n) {
-              Ok(input.take_split(index))
-            } else {
-              Err(Err::Error(Error::from_error_kind(
-                input,
-                ErrorKind::TakeWhileMN,
-              )))
-            };
-            res
+    for (i, item) in input.iter_elements().enumerate() {
+      if !cond(item) {
+        if i >= m {
+          if let Ok(index) = input.slice_index(min(i, n)) {
```

Comment:
it looks like the loop runs until the first element that does not match, but it should check that `i <=n` on every iteration. Otherwise it could get very far